src/gov/nist/javax/sip/stack/SIPDialog.java

```diff
@@ -4409,8 +4409,13 @@ public void setEarlyDialogTimeoutSeconds(int seconds) {
     public void checkRetransmissionForForking(SIPResponse response) {
         final int statusCode = response.getStatusCode();
         final String responseMethod = response.getCSeqHeader().getMethod();
-        final long responseCSeqNumber = response.getCSeq().getSeqNumber();   
-        boolean isRetransmission = !responsesReceivedInForkingCase.add(statusCode + "/" + responseCSeqNumber + "/" + responseMethod);            
+        final long responseCSeqNumber = response.getCSeq().getSeqNumber();
+        RSeq rseq = (RSeq) response.getHeader(RSeqHeader.NAME);
+        String responseRSeqNumber = "";
+        if(rseq != null) {
+            responseRSeqNumber = "/" + rseq.getSeqNumber();
+        }
+        boolean isRetransmission = !responsesReceivedInForkingCase.add(statusCode + "/" + responseCSeqNumber + "/" + responseMethod + responseRSeqNumber);
```

Comment:
So, that junit test was part of your service testsuite. I was wondering If we could add some test to JAIN-SIP test-suite...